src/os.cpp

```diff
@@ -295,9 +295,39 @@ int os_fetch_file(FILE *f, Buf *out_buf) {
     static const ssize_t buf_size = 0x2000;
     buf_resize(out_buf, buf_size);
     ssize_t actual_buf_len = 0;
+
+    bool first_read = true;
+
     for (;;) {
         size_t amt_read = fread(buf_ptr(out_buf) + actual_buf_len, 1, buf_size, f);
         actual_buf_len += amt_read;
+
+#if defined(ZIG_OS_POSIX)
+        // posix shebang for zig run
```

Comment:
`os_fetch_file` is intended to return file contents verbatim. I think it would be better to put this code outside os.cpp. This could break `@embedFile` or the code that reads a linker script, for example.

Reply:
Good point.